### PR TITLE
flatbuffers: add version 23.1.21

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **flatbuffers/23.1.21**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/flatbuffers/all/conanfile.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from conan.tools.apple import is_apple_os
 from conan.tools.build import check_min_cppstd, valid_min_cppstd
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
-from conan.tools.files import apply_conandata_patches, collect_libs, copy, get, replace_in_file, rmdir
+from conan.tools.files import export_conandata_patches, apply_conandata_patches, collect_libs, copy, get, replace_in_file, rmdir
```
Comment (SpaceIm) [inline recipes/flatbuffers/all/conanfile.py:5]: required_conan_version to 1.52.0 please

you can also take advantage of this PR to use rm_safe to remove fPIC in configure (conan 1.53.0)

Comment (toge) [reply recipes/flatbuffers/all/conanfile.py:5]: Thank a lot!
Fixed it.

**Comment by paulharris** [comment]
Why not version 23.01.21,
Or better, 2023.01.21

**Comment by toge** [comment]
@paulharris 
Thank for your comment.

I referred the flatbuffers versioning document.
https://github.com/google/flatbuffers/wiki/Versioning

Is there another document about versioning?

**Review by SpaceIm — APPROVED** [review]

**Comment by paulharris** [comment]
hrm, quote: "FlatBuffers has a version in the format: YY.MM.DD"
and then they give the example 22.9.14  which doesn't fit the traditionally accepted version of MM
ok well if that is what they say ... I guess it keeps it purely numeric.

**Review by paulharris — APPROVED** [review]

File: recipes/flatbuffers/all/conanfile.py [hunk 2]
```diff
@@ -36,20 +36,30 @@ def _has_flatc(self):
...
         copy(self, os.path.join("cmake", "FlatcTargets.cmake"), self.recipe_folder, self.export_sources_folder)
-        for p in self.conan_data.get("patches", {}).get(self.version, []):
-            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def _cmake_new_enough(self, required_version):
```
Comment (prince-chrismc) [inline recipes/flatbuffers/all/conanfile.py:45]: There was conversations within the Conan team about conditioning the graph...

That not ideal but our desired solution doesn't exist  so maybe refrain from adding this check too much pkease... after the launch of 2.0 this month we can focus the team on these issues 🤞

Comment (prince-chrismc) [reply recipes/flatbuffers/all/conanfile.py:45]: Mobile typing means terrible English 😑

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**